hcl/hcl_test.go

```diff
@@ -1100,14 +1291,23 @@ func testParser(t *testing.T, tc testcase) {
 		for _, inputConfigFile := range tc.input {
 			filename := inputConfigFile.filename
 			if filename == "" {
-				filename = config.DefaultFilename
+				panic("expect a filename in the input config")
 			}
-			cfgfile := test.WriteFile(t, configsDir, filename, inputConfigFile.body)
-			addFilenameToErrorsFileRanges(tc.want.errs, cfgfile)
+			test.WriteFile(t, configsDir, filename, inputConfigFile.body)
 		}
+		fixupFiledirOnErrorsFileRanges(configsDir, tc.want.errs)
 		got, err := hcl.ParseDir(configsDir)
 		errtest.AssertErrorList(t, err, tc.want.errs)
 
+		var gotErrs *errors.List
+		if errors.As(err, &gotErrs) {
+			if len(gotErrs.Errors()) != len(tc.want.errs) {
```

Comment:
hmm maybe we want that on AssertErrorList ? The current behavior is partial matching but not sure this is what we should have as default behavior 🤔

Reply:
yeah I think makes sense to check this in the AssertErrorList. The problem is that I got a case where tests were passing but I wasn't checking all errors coming from the parser.
This is quite useful sometimes but in this case it made me anxious haha